src/azure-cli/azure/cli/command_modules/monitor/operations/subscription_diagnostic_settings.py

```diff
@@ -22,7 +22,7 @@ def create_subscription_diagnostic_settings(cmd, client,
                                                         service_bus_rule_id=service_bus_rule,
                                                         logs=logs,
                                                         location=location)
-    return client.create_or_update(subscription_id=get_subscription_id(cmd.cli_ctx), parameters=parameters, name=name)
```

Comment:
Why delete subscription_id here ?

Reply:
Since there is no argument defined as `subscription_id` in new SDK: https://github.com/Azure/azure-sdk-for-python/blob/b5065544fa998182c13496308304d5a764261809/sdk/monitor/azure-mgmt-monitor/azure/mgmt/monitor/v2021_05_01_preview/operations/_subscription_diagnostic_settings_operations.py#L104-L109